staging/src/k8s.io/controller-manager/config/types.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LeaderMigrationConfiguration provides versioned configuration for all migrating leader locks.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type LeaderMigrationConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// LeaderName is the name of the resource under which the controllers should be run.
```

Comment:
I think we need more detail to this explanation. I believe this will be the name of the leader election resource/instance which protects the migration. Eg. "1-20-KCM-to-1-21-CCM"

Reply:
Updated a bit...hopefully it is clearer.